community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala

```diff
@@ -138,7 +138,7 @@ abstract class CompiledExecutionResult(taskCloser: TaskCloser, statement:Stateme
 
   override def notifications = Iterable.empty[InternalNotification]
 
-  override def hasNext = innerIterator.hasNext
+  override def hasNext = !taskCloser.isClosed && innerIterator.hasNext
```

Comment:
No it is not fine if there are more items to iterate, we will almost always prematurely abort the iteration.